gremlin-go/driver/traversal.go

```
@@ -96,3 +78,29 @@ func (t *Traversal) Iterate() (*Traversal, <-chan bool, error) {
 
 	return t, r, nil
 }
+
+func (t *Traversal) HasNext() (bool, error) {
+	results, err := t.getResults()
+	if err != nil {
+		return false, err
+	}
+	return !(*results).IsEmpty(), nil
+}
+
+func (t *Traversal) Next() (*Result, error) {
+	results, err := t.getResults()
+	if err != nil || (*results).IsEmpty() {
+		return nil, err
```

Should we return an error message if there isnt a value to return?

---

To be addressed in ticket AN-1051